gcc/d/d-spec.c

```diff
@@ -394,20 +404,20 @@ lang_specific_driver (cl_decoded_option **in_decoded_options,
     }
 
   /* Add `-lgphobos' if we haven't already done so.  */
-  if (library > 0 && need_phobos)
+  if (phobos_library != PHOBOS_NOLINK && need_phobos)
```

Comment:
This is a change in behavior, as `PHOBOS_DEFAULT` will now also link in the library, where it didn't before afaics. Is this intended?

Reply:
I think the behaviour is correct here. As per comment.

> libgphobos should be linked in if it is needed.

If `need_phobos` is true then it should be linked in.

Reply:
The only time when it would be `PHOBOS_DEFAULT` at this stage is were it to be passed only `.di` sources.  I don't think it's unreasonable to so, infact I could simplify logic so that it always links in libphobos if any `OPT_SPECIAL_input_file` is seen.

Reply:
ok, should be fine then.